src/main/scala/extractor/ExtractImplicits.scala

```diff
@@ -0,0 +1,72 @@
+package extractor
+
+import java.util.concurrent.ConcurrentLinkedQueue
+
+import extractor.Serializables.{Apply, DeclaredImplicit, FunApplyWithImplicitParam, ImplicitParam}
+import org.langmeta.inputs.Input
+
+import scala.meta._
+
+object ExtractImplicits {
+
+  final case class Result(
+     params: Set[ImplicitParam],
+     funs: Seq[Apply],
+     links: Set[FunApplyWithImplicitParam],
+     implicits: Set[DeclaredImplicit])
+
+  final case class LinkPair(param: ImplicitParam, fun: Apply)
+
+  def apply(ctx: SemanticCtx): Result = {
+    val file: String = ctx.input match {
+      case Input.VirtualFile(path, _) => path
+      case Input.File(path, _) => path.toString
+      case _ => ""
+    }
+
+    lazy val syntheticImplicits =
```

Comment:
the for should go at the end of ` =` to make it more compact and consistent with match